app/views/schedules/_task_fields.html.erb

```diff
@@ -19,11 +21,14 @@
                      class: 'form-control' %>
 
         <span class="input-group-btn">
-        <%= link_to_remove_association '',
-                                       f,
-                                       alt: 'Remove Task',
-                                       title: 'Remove Task',
-                                       class: 'btn btn-danger glyphicon glyphicon-minus' %>
+        <% unless !f.object.schedule.nil? and f.object.id == f.object.schedule.root_task_id %>
```

Comment:
Rather than making this view more complicated, you should add fields for the Task under `app/views/schedules/_fields.html.erb`.